tests/ssl-opt.sh

```diff
@@ -3888,6 +3888,20 @@ run_test    "Small packet TLS 1.2 AEAD shorter tag" \
             0 \
             -s "Read from client: 1 bytes read"
 
+# Test for zero-length application data messages with CBC
+# (using a CBC cipher here causes a zero length application data
+# message to be sent (16 padding bytes.) This should be ignored
+# by mbedTLS and not returned as data to the caller
+#
+# (The actual bad data is 16 repeated 0x0f bytes, which can't
+# be matched by cross-platform grep.)
```

Comment:
I spent a lot of time trying to write a better test for this (actually checking for 0x0f in the output, rather than just a length). The better idea I had is to change run_test so it will fail if any test *ever* outputs a non-printable character to stdout. However I couldn't find a way to do check this with a cross-platform grep. Maybe there are some other programs which are guaranteed available in the test environment?

Reply:
I think the test is OK as it stands, though if you want to use python to mangle the output so that you can then grep for `\x0f` that's fine too (we're already using python3 elsewhere in our CI, so this wouldn't be a new dependency).

However I have a bigger issue with the test: it doesn't really test anything with OpenSSL 1.0.2g (currently the default version in our CI, still based on Ubuntu 16.04) as this version doesn't negociate encrypt-then-MAC (and the problem only occurs with EtM). Fortunately, our CI also has a recent enough OpenSSL available.

In order to make sure this test is also run with a meaningful OpenSSL version, please add the following line in `tests/scripts/all.sh` just below line 472:
```
if_build_succeeded env OPENSSL_CMD="$OPENSSL_NEXT" tests/ssl-opt.sh -f 'Zero-length application data'
```

Reply:
Actually I changed my mind. Rather than relying on a specific OpenSSL version, it seems more robust to just test with ourselves, as @andresag01 made possible with #1852. So I would now suggest removing the test from this PR, as the test will be added in #1852 (to be merge right after this one).

Reply:
Hi @mpg, ah yes, this is a much better test. Will remove here.

Reply:
(BTW, regarding the test environment in general I have an open question about what the standard environment is in #1568. Any feedback there (or a small section in the readme or developer docs) would be really appreciated!)